### PR TITLE
log: update crdbV2 log decoder

### LINKED ISSUE #104210: Debug Zip Missing Panic Logs
**Describe the problem**

I collected a debug zip for a cluster that recently had nodes restart due to a panic. I was taking the debug zip in order to determine why a node was panicing. The stderr log containing the unexpected panic was not included in the debug zip. 

The debug zip logs contained this error that was generated by taking the debug zip. The error seems to suggest CRDB was unable to parse the stderr logs including the panic:

```
E230601 14:51:13.619603 86574 server/status.go:1310 ⋮ [T1,n1,status] 581 decoding on line 8: malformed log entry
(1) attached stack trace
  -- stack trace:
  | github.com/cockroachdb/cockroach/pkg/util/log.(*entryDecoderV2).Decode.func1
  | 	github.com/cockroachdb/cockroach/pkg/util/log/format_crdb_v2.go:528
  | runtime.gopanic
  | 	GOROOT/src/runtime/panic.go:884
  | [...repeated from below...]
Wraps: (2) decoding on line 8
Wraps: (3) attached stack trace
  -- stack trace:
  | github.com/cockroachdb/cockroach/pkg/util/log.(*entryDecoderV2).peekNextFragment
  | 	github.com/cockroachdb/cockroach/pkg/util/log/format_crdb_v2.go:615
  | github.com/cockroachdb/cockroach/pkg/util/log.(*entryDecoderV2).Decode
  | 	github.com/cockroachdb/cockroach/pkg/util/log/format_crdb_v2.go:548
  | github.com/cockroachdb/cockroach/pkg/server.(*statusServer).LogFile
  | 	github.com/cockroachdb/cockroach/pkg/server/status.go:1306
  | github.com/cockroachdb/cockroach/pkg/server/serverpb._Status_LogFile_Handler.func1
  | 	github.com/cockroachdb/cockroach/pkg/server/serverpb/bazel-out/k8-opt/bin/pkg/server/serverpb/serverpb_go_proto_/github.com/cockroachdb/cockroach/pkg/server/serverpb/status.pb.go:9040
  | github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor.ServerInterceptor.func1
  | 	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:115
  | google.golang.org/grpc.chainUnaryInterceptors.func1.1
  | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1162
  | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func3
  | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:276
  | google.golang.org/grpc.chainUnaryInterceptors.func1.1
  | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1165
  | github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor
  | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/auth.go:105
  | google.golang.org/grpc.chainUnaryInterceptors.func1.1
  | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1165
  | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1.1
  | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:243
  | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
  | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:322
  | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1
  | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:241
  | google.golang.org/grpc.chainUnaryInterceptors.func1.1
  | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1165
  | google.golang.org/grpc.chainUnaryInterceptors.func1
  | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1167
  | github.com/cockroachdb/cockroach/pkg/server/serverpb._Status_LogFile_Handler
  | 	github.com/cockroachdb/cockroach/pkg/server/serverpb/bazel-out/k8-opt/bin/pkg/server/serverpb/serverpb_go_proto_/github.com/cockroachdb/cockroach/pkg/server/serverpb/status.pb.go:9042
  | google.golang.org/grpc.(*Server).processUnaryRPC
  | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1340
  | google.golang.org/grpc.(*Server).handleStream
  | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1713
  | google.golang.org/grpc.(*Server).serveStreams.func1.2
  | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:965
  | runtime.goexit
  | 	GOROOT/src/runtime/asm_amd64.s:1594
Wraps: (4) malformed log entry
Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *withstack.withStack (4) *errutil.leafError
```

The content of the std error file was:

```
sh-4.4# cat cockroach-stderr.cockroachdb-0.root.2023-05-31T21_52_20Z.000030.log
I230531 21:52:20.499220 1 util/log/file_sync_buffer.go:238 ⋮ [config]   file created at: 2023/05/31 21:52:20
I230531 21:52:20.499241 1 util/log/file_sync_buffer.go:238 ⋮ [config]   running on machine: cockroachdb-0
I230531 21:52:20.499248 1 util/log/file_sync_buffer.go:238 ⋮ [config]   binary: CockroachDB CCL v23.1.2 (x86_64-pc-linux-gnu, built 2023/05/25 15:51:13, go1.19.4)
I230531 21:52:20.499255 1 util/log/file_sync_buffer.go:238 ⋮ [config]   arguments: [/cockroach/cockroach start --certs-dir /cockroach/cockroach-certs --listen-addr 0.0.0.0:26258 --http-addr 0.0.0.0:8080 --sql-addr 0.0.0.0:26257 --advertise-addr cockroachdb-0.cockroachdb.us-west-2.svc.cluster.local:26258 --advertise-sql-addr cockroachdb-0.cockroachdb.us-west-2.svc.cluster.local:26257 --join _grpc._tcp.cockroachdb.us-west-2.svc --experimental-dns-srv --cache 25% --max-sql-memory 25% --locality=region=aws-us-west-2,az=aws-us-west-2a,dns=cockroachdb-0.cockroachdb.us-west-2.svc.cluster.local --log-config-file /cockroach/config/logs.yaml --log-config-vars HOST_IP --external-io-dir=disabled --max-offset=100ms]
I230531 21:52:20.499284 1 util/log/file_sync_buffer.go:238 ⋮ [config]   log format (utf8=✓): crdb-v2
I230531 21:52:20.499288 1 util/log/file_sync_buffer.go:238 ⋮ [config]   line format: [IWEF]yymmdd hh:mm:ss.uuuuuu goid [chan@]file:line redactionmark \[tags\] [counter] msg
I230531 21:52:20.498414 1 util/log/flags.go:211  [-] 1  stderr capture started
panic: invalid tenant ID 0

goroutine 6603729 [running]:
github.com/cockroachdb/cockroach/pkg/roachpb.assertValid(0x5398fe0?)
        github.com/cockroachdb/cockroach/pkg/roachpb/pkg/roachpb/tenant.go:99 +0x59
github.com/cockroachdb/cockroach/pkg/roachpb.MustMakeTenantID(...)
        github.com/cockroachdb/cockroach/pkg/roachpb/pkg/roachpb/tenant.go:52
github.com/cockroachdb/cockroach/pkg/ccl/backupccl.backupShowerDefault.func1({0x7281e90, 0xc00644dd40}, {{0xc008bfee10, 0x2e0}, {0xc00c389130, 0x1, 0x1}, {0xc001812600, 0x1, 0x1}, ...})
        github.com/cockroachdb/cockroach/pkg/ccl/backupccl/show.go:967 +0x24eb
github.com/cockroachdb/cockroach/pkg/ccl/backupccl.manifestInfoReader.showBackup({{{0xc008961500?, 0x0?, 0x0?}, 0xc00c284120?}}, {0x7281e90?, 0xc00c284fc0?}, _, _, {{0xc008bfee10, 0x2e0}, ...}, ...)
        github.com/cockroachdb/cockroach/pkg/ccl/backupccl/show.go:120 +0x205
github.com/cockroachdb/cockroach/pkg/ccl/backupccl.showBackupPlanHook.func1({0x7281e90, 0xc00c284f00}, {0x0?, 0x0?, 0x0?}, 0xc00e8503c0?)
        github.com/cockroachdb/cockroach/pkg/ccl/backupccl/show.go:531 +0x1fc5
github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1()
        github.com/cockroachdb/cockroach/pkg/sql/planhook.go:205 +0xa4
created by github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec
        github.com/cockroachdb/cockroach/pkg/sql/planhook.go:203 +0x1b2
```

Epic CRDB-32134

### PR DESCRIPTION
**Pull request by aa-joshi**
Previously, we were returning an error during decode when there is a malformed log entry. We were ignoring unstructured log line & returning  error. This was causing issues during debug zip generation. To address this, we have updated the crdbV2 log decoder so that it would return malformed/unstructured line as log entry along with error. LogFile RPC would append such logs along with error.

Fixes: #104210
Release note (bug fix): This patch fixes the bug where debug zip is giving error while fetching unstructured/malformed logs.

### CODE REVIEW

**Review by dhartunian — COMMENTED** [review]
> Reviewed 1 of 5 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/127057)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @aa-joshi, @arjunmahishi, and @xinhaoz)

___
*[`-- commits` line 2 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/127057#-O1qjyeW8kTHfMonFD-A):*
This title is too general. I expect something like `log: crdbV2 decoder handles malformed lines`
___
*[`pkg/cli/zip_per_node.go` line 468 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/127057#-O1qkfAyCgT5lwRgc-M-:-O1qkfAyCgT5lwRgc-M0:b-5rv4xl) ([raw file](https://github.com/cockroachdb/cockroach/blob/a22d95611ad8f706f8233fb7dc22c9352338b53b/pkg/cli/zip_per_node.go#L468)):*
> ```Go
> 				}
> 			}
> 			sf = logPrinter.start("writing output")
> ```

Why does this start a new unit of work in the logPrinter instead of calling `.progress` as before?
___
*[`pkg/util/log/format_crdb_v2.go` line 506 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/127057#-O1qu3zrCQEDHF6EZwAQ:-O1qu3zrCQEDHF6EZwAR:b-njjpej) ([raw file](https://github.com/cockroachdb/cockroach/blob/a22d95611ad8f706f8233fb7dc22c9352338b53b/pkg/util/log/format_crdb_v2.go#L506)):*
> ```Go
> 
> // Decode decodes the next log entry into the provided protobuf message.
> func (d *entryDecoderV2) Decode(entry *logpb.Entry) (err error) {
> ```

Edit docstring to explain how malformed lines are handled and what the caller's responsibility is.
___
*[`pkg/util/log/format_crdb_v2.go` line 526 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/127057#-O1qycpqDbPFtIga6pnp:-O1qycpqDbPFtIga6pnq:bx2y59x) ([raw file](https://github.com/cockroachdb/cockroach/blob/a22d95611ad8f706f8233fb7dc22c9352338b53b/pkg/util/log/format_crdb_v2.go#L526)):*
> ```Go
> 			//Construct log entry based on unstructured line.
> 			entry.Message = string(d.malformedLine)
> 		}
> ```

Can you treat the malformed line as a fragment? These two clauses are confusing because they do similar but not quite the same things in an `err ==nil` and `err != nil` case which is confusing.



_Code quote:_
```Go
	//Construct the log from unstructured line which was identified in
	//log continuation.
	if err == nil && len(frag) == 0 {
		entry.Message = string(d.malformedLine)
		d.popFragment()
		return err
	}

	if err != nil {
		if errors.Is(err, ErrMalformedLogEntry) {
			//Construct log entry based on unstructured line.
			entry.Message = string(d.malformedLine)
		}
```
___
*[`pkg/util/log/format_crdb_v2.go` line 545 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/127057#-O1qynoo8uTZmr9yeSjs:-O1qynoo8uTZmr9yeSjt:b-z42vjb) ([raw file](https://github.com/cockroachdb/cockroach/blob/a22d95611ad8f706f8233fb7dc22c9352338b53b/pkg/util/log/format_crdb_v2.go#L545)):*
> ```Go
> 				//Treat unstructured log entry as empty fragment so that next
> 				//peekNextFragment call would construct log based on it.
> 				d.nextFragment = entryDecoderV2Fragment{}
> ```

Similar to above comment, would be nice to avoid this slight hack. Maybe it's worth transporting the malformed line inside the error? I'm not sure.

As written this is hard to follow because the first time you see malformed line it's because it was a "continuation" of a well-formed line that we hack `Decode` into using.

**Review by aa-joshi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/127057)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @arjunmahishi, @dhartunian, and @xinhaoz)

___
*[`-- commits` line 2 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/127057#-O1qjyeW8kTHfMonFD-A):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

This title is too general. I expect something like `log: crdbV2 decoder handles malformed lines`
</blockquote></details>

Done.
___
*[`pkg/cli/zip_per_node.go` line 468 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/127057#-O1qkfAyCgT5lwRgc-M-:-O2-P7J_41JAfGolxVYz:b-iioirs) ([raw file](https://github.com/cockroachdb/cockroach/blob/a22d95611ad8f706f8233fb7dc22c9352338b53b/pkg/cli/zip_per_node.go#L468)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

Why does this start a new unit of work in the logPrinter instead of calling `.progress` as before?
</blockquote></details>

It is a bug fix. When we encounter parsing error we are marking reporter `done` as part of `createError`. We can't call progress on closed reporter. Hence, we are starting a new for writing output. Please note that it is invoked on same `logPrinter` to make sure that we are tracking for same log file.
https://github.com/cockroachdb/cockroach/blob/dca553c77b8d9a2a8a6b9bfc3ead1b1bd2f4babb/pkg/cli/zip_per_node.go#L467
___
*[`pkg/util/log/format_crdb_v2.go` line 506 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/127057#-O1qu3zrCQEDHF6EZwAQ:-O2-PzCS761rMvdSJ8Zz:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a22d95611ad8f706f8233fb7dc22c9352338b53b/pkg/util/log/format_crdb_v2.go#L506)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

Edit docstring to explain how malformed lines are handled and what the caller's responsibility is.
</blockquote></details>

Done.
___
*[`pkg/util/log/format_crdb_v2.go` line 526 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/127057#-O1qycpqDbPFtIga6pnp:-O2-Q1LA08PVfvWCJLP6:ba97jrf) ([raw file](https://github.com/cockroachdb/cockroach/blob/a22d95611ad8f706f8233fb7dc22c9352338b53b/pkg/util/log/format_crdb_v2.go#L526)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

Can you treat the malformed line as a fragment? These two clauses are confusing because they do similar but not quite the same things in an `err ==nil` and `err != nil` case which is confusing.
</blockquote></details>

Done. We are treating the malformed log entry as fragment now.
___
*[`pkg/util/log/format_crdb_v2.go` line 545 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/127057#-O1qynoo8uTZmr9yeSjs:-O2-Q8AqDeowvPPyYcYy:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a22d95611ad8f706f8233fb7dc22c9352338b53b/pkg/util/log/format_crdb_v2.go#L545)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

Similar to above comment, would be nice to avoid this slight hack. Maybe it's worth transporting the malformed line inside the error? I'm not sure.

As written this is hard to follow because the first time you see malformed line it's because it was a "continuation" of a well-formed line that we hack `Decode` into using.
</blockquote></details>

Done.
___
*[`pkg/util/log/format_crdb_v2.go` line 638 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/127057#-O2-QD6B1VqeHsALb_0l:-O2-QD6B1VqeHsALb_0m:b-7v5s97) ([raw file](https://github.com/cockroachdb/cockroach/blob/a22d95611ad8f706f8233fb7dc22c9352338b53b/pkg/util/log/format_crdb_v2.go#L638)):*
> ```Go
> func (d *entryDecoderV2) initEntryFromFirstLine(
> 	entry *logpb.Entry, m entryDecoderV2Fragment,
> ) (err error) {
> ```

We are removing it as it is creating unnecessary error handling as we are not emitting any error as part of it.

**Review by arjunmahishi — COMMENTED** [review]
Minor comments

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/127057#-:-O20DyJF-u_fcr6zOoRa:b-v9qsv0)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @aa-joshi, @dhartunian, and @xinhaoz)

___
*[`pkg/util/log/format_crdb_v2.go` line 529 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/127057#-O20DCpG-nILjOl63gHM:-O20DCpG-nILjOl63gHN:b-ado4r2) ([raw file](https://github.com/cockroachdb/cockroach/blob/0b918d1dc3a9ce1f04975202be3b04ec375a816e/pkg/util/log/format_crdb_v2.go#L529)):*
> ```Go
> 		if err != nil || !frag.isContinuation() {
> 			// Ignore this error as it is relevant to the next line and we don't
> 			// know if it is continuation line or not.
> ```

This comment seems helpful. We can keep it?
___
*[`pkg/util/log/format_crdb_v2.go` line 531 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/127057#-O20CheZDbTMR94ytIUd:-O20CheZDbTMR94ytIUe:b-dbd03s) ([raw file](https://github.com/cockroachdb/cockroach/blob/55e8afcd6b2a3836684e056ff50eb7c1c41c0249/pkg/util/log/format_crdb_v2.go#L531)):*
> ```Go
> 
> 	d.popFragment()
> 	d.initEntryFromFirstLine(entry, frag)
> ```

I think this returns an error? 
I can see that the error was handled in the previous version of the code.

**Review by dhartunian — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/127057)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @aa-joshi and @xinhaoz)

___
*[`pkg/util/log/format_crdb_v2.go` line 523 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/127057#-O20vOKw52Rse3psjb9C:-O20vOKw52Rse3psjb9D:b-x99tyx) ([raw file](https://github.com/cockroachdb/cockroach/blob/55e8afcd6b2a3836684e056ff50eb7c1c41c0249/pkg/util/log/format_crdb_v2.go#L523)):*
> ```Go
> 	// construct the log entry from malformed fragment and return it along
> 	// with malformed log entry error
> 	if d.isMalformedFragment {
> ```

I think `isMalformedFragment` can be returned with the `peek` call so you don't need to manage its state separately and reset it in the `pop` call.

Given your changes in the `peek` call, I think all the code in this `if` clause will happen below if you follow the happy path of execution so you don't need to duplicate the `pop`, `init`, and `Message` field set here.

The only special case you should need is to return the `ErrMalformedLogEntry` at the end if we know the fragment is malformed.

**Review by aa-joshi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/127057)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @arjunmahishi, @dhartunian, and @xinhaoz)

___
*[`pkg/util/log/format_crdb_v2.go` line 523 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/127057#-O20vOKw52Rse3psjb9C:-O23WflpAj7zFaOlSrYP:bx6z1ki) ([raw file](https://github.com/cockroachdb/cockroach/blob/55e8afcd6b2a3836684e056ff50eb7c1c41c0249/pkg/util/log/format_crdb_v2.go#L523)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

I think `isMalformedFragment` can be returned with the `peek` call so you don't need to manage its state separately and reset it in the `pop` call.

Given your changes in the `peek` call, I think all the code in this `if` clause will happen below if you follow the happy path of execution so you don't need to duplicate the `pop`, `init`, and `Message` field set here.

The only special case you should need is to return the `ErrMalformedLogEntry` at the end if we know the fragment is malformed.
</blockquote></details>

Consider below generic use case: valid log line followed by invalid log line. Invalid log line would get parsed at   https://github.com/cockroachdb/cockroach/blob/46854026bc11b3870c8bbefc976f5550b18dd82c/pkg/util/log/format_crdb_v2.go#L526

Here, we need to make sure that valid log line is parsed into log entry & create an indication that subsequent fragment (invalid log line) is malformed. That we are doing through `isMalformedFragment`.

This is needed because in the subsequent decode call we don't know whether the fragment is valid or invalid.

We can pair for it if it is still confusing.
___
*[`pkg/util/log/format_crdb_v2.go` line 529 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/127057#-O20DCpG-nILjOl63gHM:-O23W4Wb3Po7p8ZDCXdO:b-ck04xg) ([raw file](https://github.com/cockroachdb/cockroach/blob/0b918d1dc3a9ce1f04975202be3b04ec375a816e/pkg/util/log/format_crdb_v2.go#L529)):*
<details><summary><i>Previously, arjunmahishi (Arjun Mahishi) wrote…</i></summary><blockquote>

This comment seems helpful. We can keep it?
</blockquote></details>

Make sense! Done.
___
*[`pkg/util/log/format_crdb_v2.go` line 531 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/127057#-O20CheZDbTMR94ytIUd:-O23VrFp8_fHqEhvRZvt:b-viqaql) ([raw file](https://github.com/cockroachdb/cockroach/blob/55e8afcd6b2a3836684e056ff50eb7c1c41c0249/pkg/util/log/format_crdb_v2.go#L531)):*
<details><summary><i>Previously, arjunmahishi (Arjun Mahishi) wrote…</i></summary><blockquote>

I think this returns an error? 
I can see that the error was handled in the previous version of the code.
</blockquote></details>

`initEntryFromFirstLine` was always returning error as `nil`. I have removed it so that we don't have to unnecessary add checks around error.
ref: https://reviewable.io/reviews/cockroachdb/cockroach/127057#-O2-QD6B1VqeHsALb_0l

**Review by dhartunian — COMMENTED** [review]
> Reviewed 1 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/127057)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @aa-joshi, @arjunmahishi, and @xinhaoz)

___
*[`pkg/util/log/format_crdb_v2.go` line 523 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/127057#-O20vOKw52Rse3psjb9C:-O25DerF7NO8mkv72-UJ:bub4ky1) ([raw file](https://github.com/cockroachdb/cockroach/blob/55e8afcd6b2a3836684e056ff50eb7c1c41c0249/pkg/util/log/format_crdb_v2.go#L523)):*
<details><summary><i>Previously, aa-joshi (Akshay Joshi) wrote…</i></summary><blockquote>

Consider below generic use case: valid log line followed by invalid log line. Invalid log line would get parsed at   https://github.com/cockroachdb/cockroach/blob/46854026bc11b3870c8bbefc976f5550b18dd82c/pkg/util/log/format_crdb_v2.go#L526

Here, we need to make sure that valid log line is parsed into log entry & create an indication that subsequent fragment (invalid log line) is malformed. That we are doing through `isMalformedFragment`.

This is needed because in the subsequent decode call we don't know whether the fragment is valid or invalid.

We can pair for it if it is still confusing.
</blockquote></details>

I understand the scenario, thanks for explaining it. Can you add a test case for it?

If I comment out this code all the tests still pass, unless I'm missing something.
```
	//if d.isMalformedFragment {
	//	d.popFragment()
	//	d.initEntryFromFirstLine(entry, frag)
	//	entry.Message = string(frag.getMsg())
	//	return ErrMalformedLogEntry
	//}
```

**Review by aa-joshi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/127057)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @arjunmahishi, @dhartunian, and @xinhaoz)

___
*[`pkg/util/log/format_crdb_v2.go` line 523 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/127057#-O20vOKw52Rse3psjb9C:-O28Im2ZB_dfXj6kO-rH:b99aize) ([raw file](https://github.com/cockroachdb/cockroach/blob/55e8afcd6b2a3836684e056ff50eb7c1c41c0249/pkg/util/log/format_crdb_v2.go#L523)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

I understand the scenario, thanks for explaining it. Can you add a test case for it?

If I comment out this code all the tests still pass, unless I'm missing something.
```
	//if d.isMalformedFragment {
	//	d.popFragment()
	//	d.initEntryFromFirstLine(entry, frag)
	//	entry.Message = string(frag.getMsg())
	//	return ErrMalformedLogEntry
	//}
```
</blockquote></details>

We were just printing log message both valid & invalid log entries as part of test case: https://github.com/cockroachdb/cockroach/blob/59be0708627bb27f49d8976182062c6886d64282/pkg/util/log/format_crdb_v2_test.go#L270
I have updated the test now to differentiate between log scenarios.
Thanks for pointing it out! PTAL.

**Review by dhartunian — APPROVED** [review]
:lgtm: thx for the changes

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/127057#-:-O2P8CIzDshucVVxLxBB:br34dsi)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @arjunmahishi and @xinhaoz)

**Comment by aa-joshi** [comment]
TFTR!
